cai_causal_graph/time_series_causal_graph.py

```diff
@@ -0,0 +1,888 @@
+                variable_name is not None and time_lag is not None
+            ), 'Either identifier or (variable_name and time_lag) must be provided.'
+
+        node = TimeSeriesNode(
+            identifier=identifier,
+            time_lag=time_lag,
+            variable_name=variable_name,
+            variable_type=variable_type,
+            meta=meta,
+        )
+        identifier = node.identifier
+        if identifier in self._nodes_by_identifier:
+            raise CausalGraphErrors.NodeDuplicatedError(f'Node already exists: {identifier}')
+
+        self._nodes_by_identifier[identifier] = node
+
+        return node
+
+    @_reset_attributes
+    def replace_node(
+        self,
+        /,
+        node_id: NodeLike,
+        new_node_id: Optional[NodeLike] = None,
+        time_lag: Optional[int] = None,
+        variable_name: Optional[str] = None,
+        *,
+        variable_type: NodeVariableType = NodeVariableType.UNSPECIFIED,
+        meta: Optional[dict] = None,
+    ):
+        """
+        Replace a node in the graph.
+
+        See `cai_causal_graph.causal_graph.CausalGraph.replace_node` for more details.
+        """
+        # either new_node_id or (variable_name and time_lag) must be provided
+        assert new_node_id is not None or (
+            variable_name is not None and time_lag is not None
+        ), 'Either new_node_id or (variable_name and time_lag) must be provided.'
+
+        if new_node_id is None:
+            # update name from the variable name and time lag
+            new_node_id = get_name_with_lag(variable_name, time_lag)   # type: ignore
+        else:
+            new_node_id = Node.identifier_from(new_node_id)
+            variable_name, time_lag = get_variable_name_and_lag(Node.identifier_from(new_node_id))
+
+        if meta is not None:
+            meta = meta.copy()
+            meta.update({VARIABLE_NAME: variable_name, TIME_LAG: time_lag})
+        else:
+            meta = {VARIABLE_NAME: variable_name, TIME_LAG: time_lag}
+
+        super().replace_node(node_id, new_node_id, variable_type=variable_type, meta=meta)
+
+    @_reset_attributes
+    def delete_node(self, identifier: NodeLike):
+        """
+        Delete a node from the graph.
+
+        See `cai_causal_graph.causal_graph.CausalGraph.delete_node` for more details.
+        """
+        super().delete_node(identifier)
+
+    @_reset_attributes
+    def delete_edge(self, source: NodeLike, destination: NodeLike):
+        """
+        Delete an edge from the graph.
+
+        See `cai_causal_graph.causal_graph.CausalGraph.delete_edge` for more details.
+        """
+        super().delete_edge(source, destination)
+
+    @_reset_attributes
+    def add_edge(
+        self,
+        /,
+        source: Optional[NodeLike] = None,
+        destination: Optional[NodeLike] = None,
+        *,
+        edge_type: EDGE_T = EDGE_T.DIRECTED_EDGE,
+        meta: Optional[dict] = None,
```

Comment:
I would remove this parameter `meta`, as I don't see its purpose.

Reply:
It is used downstream for modeling.